controllers/machine.go

```diff
@@ -326,6 +328,27 @@ func (mrc *machineReconcileContext) ensureHardwareUserData(hardware *tinkv1.Hard
 	return nil
 }
 
+func (mrc *machineReconcileContext) ensureHardwareDiskData(hardware *tinkv1.Hardware) error {
+       foundDisks := make([]tinkv1.Disk, 0)
+       metadata := hardware.Spec.Metadata
+       if metadata != nil {
+			instances := metadata.Instance
+			if instances != nil {
+					storage := metadata.Instance.Storage
+					if storage != nil {
```

Comment:
I think you can make this a single `if` block
```suggestion
       if hardware.Spec.Metadata != nil && hardware.Spec.Metadata.Instance  != nil && hardware.Spec.Metadata.Instance.Storage  != nil {
```